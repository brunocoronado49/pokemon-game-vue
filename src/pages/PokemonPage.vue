<template>
<h1>Who is this Pokemon?</h1>
<PokemonPicture :pokemonId="5" :showPokemon="false" />
<PokemonOptions />
</template>

<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'
import getPokemonOptions from '@/helpers/getPokemonOptions'

console.log(getPokemonOptions())

export default {
    name: 'PokemonPage',
    components: {
        PokemonOptions,
        PokemonPicture
   }
}
</script>
