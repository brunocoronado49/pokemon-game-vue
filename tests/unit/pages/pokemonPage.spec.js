import { shallowMount, mount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'

import { pokemonsMock } from '../mocks/pokemons.mock'

describe('PokemonPage test', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
    })

    test('should do match woth the snaopshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('should call mixPokemonArr in the mount', () => {
        const mixPokemonArrSpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount(PokemonPage)
        expect(mixPokemonArrSpy).toHaveBeenCalled()
    })

    test('should match with snapshot when pokemons are loaded', () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: [],
                    pokemon: pokemonsMock[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })
})

