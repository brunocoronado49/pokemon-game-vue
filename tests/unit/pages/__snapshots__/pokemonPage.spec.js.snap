// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage test should do match woth the snaopshot 1`] = `
<div>
  <h1>Cargando...</h1>
</div>
`;

exports[`PokemonPage test should match with snapshot when pokemons are loaded 1`] = `
<div>
  <h1>¿Quién es este Pokémon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons=""></pokemon-options-stub>
  <!--v-if-->
</div>
`;
